src/fromager/resolver.py

```diff
@@ -523,9 +528,38 @@ def __init__(
         version_source: VersionSource,
         constraints: Constraints | None = None,
         req_type: RequirementType | None = None,
+        matcher: MatchFunction | re.Pattern | None = None,
     ):
         super().__init__(constraints=constraints, req_type=req_type)
         self._version_source = version_source
+        if matcher is None:
+            self._match_function = self._default_match_function
+        elif isinstance(matcher, re.Pattern):
+            self._match_function = functools.partial(
+                self._re_match_function, regex=matcher
+            )
+        else:
+            self._match_function = matcher
+
+    def _default_match_function(self, identifier: str, item: str) -> Version | None:
+        try:
+            return Version(item)
+        except Exception as err:
+            logger.debug(f"{identifier}: could not parse version from {item}: {err}")
+            return None
+
+    def _re_match_function(
+        self, identifier: str, item: str, *, regex: re.Pattern
+    ) -> Version | None:
+        mo = regex.match(item)
+        if mo is None:
+            return None
```

Comment:
It would help to log tags that fail to match the regex. I had to add that locally when debugging an issue with a plugin earlier this week.

Reply:
Good point, it's an easy fix!